### PR TITLE
Fix #320 'read' value is ignored and creates errors

### LINKED ISSUE #320: `@ContentChildren` ignores `read` parameter and therefore complains about valid type
image

Probably similar with `@ContentChild()`, `@ViewChildren()`, `@ViewChild()`

### PR DESCRIPTION
**Pull request by MichaelRFairhurst**

### CODE REVIEW

File: analyzer_plugin/lib/src/directive_linking.dart [hunk 1]
```diff
@@ -381,18 +381,35 @@ class ContentChildLinker {
     }
   }
 
+  /// See [getFieldWithInheritance]
+  DartObject getSelectorWithInheritance(DartObject value) =>
+      getFieldWithInheritance(value, "selector");
```
Comment (mk13) [inline analyzer_plugin/lib/src/directive_linking.dart]: Prefer single quote.

File: analyzer_plugin/lib/src/directive_linking.dart [hunk 2]
```diff
@@ -381,18 +381,35 @@ class ContentChildLinker {
...
+
+  /// See [getFieldWithInheritance]
+  String getReadWithInheritance(DartObject value) {
+    final constantVal = getFieldWithInheritance(value, "read");
```
Comment (mk13) [inline analyzer_plugin/lib/src/directive_linking.dart]: Also prefer single comment here.

File: analyzer_plugin/lib/src/directive_linking.dart [hunk 3]
```diff
@@ -381,18 +381,35 @@ class ContentChildLinker {
...
   /// until either it ends, or we find a "selector" field.
-  DartObject getSelectorWithInheritance(DartObject value) {
-    final selector = value.getField("selector");
+  DartObject getFieldWithInheritance(DartObject value, String field) {
+    final selector = value.getField(field);
     if (selector != null) {
       return selector;
     }
 
     final _super = value.getField("(super)");
```
Comment (mk13) [inline analyzer_plugin/lib/src/directive_linking.dart]: Here too.

File: analyzer_plugin/test/angular_driver_test.dart [hunk 4]
```diff
@@ -2686,6 +2686,35 @@ class ContentChildComp {}
...
+
+@Component(selector: 'foo', template: '')
+class ContentChildComp {}
+''';
+    final source = newSource('/test.dart', code);
+    await getViews(source);
+    final component = directives.first;
+    final childrens = component.contentChildren;
+    expect(childrens, hasLength(1));
+    expect(childrens.first.read, equals("ViewContainerRef"));
```
Comment (mk13) [inline analyzer_plugin/test/angular_driver_test.dart]: Also here!

File: analyzer_plugin/test/angular_driver_test.dart [hunk 5]
```diff
@@ -2686,6 +2686,35 @@ class ContentChildComp {}
...
+    final childs = component.contentChildren;
+    expect(childs, hasLength(1));
+    expect(childs.first.read, equals("ViewContainerRef"));
```
Comment (mk13) [inline analyzer_plugin/test/angular_driver_test.dart]: Last one!

**Comment by mk13** [comment]
Sorry for being nit-picky, but after your omega clean up, I feel like it would be good to continue adhering to strong code convention.

**Comment by MichaelRFairhurst** [comment]
No, good catches, and agreed!

Wish there was a lint for this...

**Pull request merged**